Bottom: c5372b6b604a9f3cb020ef06ea46a9769b6819a4
Top:    90dfaaac359e90ed963edf0ad755e1bdeb2cb9e5
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-31 04:20:13 +0000

Workaround crash where contour is passed as NULL to poly_PreContour









---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
